docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+be saturated firstly and combating the generic DDoS is out of scope for this
+proposal.
+
+#### Stalling a block production via turbine by leaders
+
+This attack can be mounted in the hopes of reordering transactions after
+descrambling. However, this is prevented by the very short wall-clock based
+expiration attached to scrambled transactions. Also, pre-vote is equipped with
+wall clock to ignore the whole block if this particular opt-conf duration for
+the block is longer than the target duration of scrambling.
+
+#### Stalling revealed key propagation via turbine by leaders
+
+Naturally, leaders want their blocks to be included in the major fork. However,
+malice leader can still annoy the cluster by stalling the descrambling phase
+in this way. To mitigate this, we allow subsequent leaders can also propagate
+the revealed keys shreds.
+
+#### Selfish mining by only including theirs own MEV transactions
+
+the node must be smarter than all, otherwise reward will be less.
+
+also, may impose 1/3+ vote instruction in block not to be skipped.
+
+#### Spamming with garbage scrambles by leader or (malicious) validators
+
+- generally should be fine. the leader just caps the max txes in a block
+ - mining bad scramble reward less or none? (forced and chunked descrabmling will be needed for proper rewarding system in that case)
+
+#### Stalling the banking stage after full revelation of scrambled transactions
+
+Likewise above, the mitigation here is that the next leader can replay the
+previous leader's block if the previous leader has silenced.
+
+### Optimal behaviors for each role
+
+#### Clients
+
+They don't want to be targeted by MEV for any trading transactions.
+Thereby they scramble the transaction and never share the derived private key to
+descramble. Also, they can't withhold revelation of their scrambled transaction
+as its outcome turned out unfavorable. Also, they want to protect their
+transactions from cluster stalls. So, they make transactions quickly expire.
+Thus, their strategy for most likely favorable trade is submission of their
+transaction as fast as possible as determined.
+
+Lastly, transactions are now included into a block very greedily, meaning the
+5000 lamports fee will be debited once broadcast (and included into a block).
+And observation by the cluster can be signalled very timely and very reliably
+to that client. So, clients (esp.  bots) are less incentivised to spamming for
+the race of inclusion.
+
+#### Leaders
+
+They want to maximize profit from running the validator in any possible ways.
+So, they try to stall the block production for MEV with the help of searchers.
+However, stalling block production is guaranteed to result in skipped block. All
+in all, this behavior incurs opportunity cost. Thus, their optimal behavior is
+include transactions indiscriminately as many as possible for maximum block
+rewards. This naturally results in the FIFO ordering of transactions.
+
+#### Validators
```

Comment:
> What are the insentives for validators to actually permit this functionality ? Because MEV seems like a decent insentive to opt out of it, validators could just not accept scrambled transactions (build empty blocks for descrambling).

your concern is correct. considering the already existing mev ecosystem (jito), it's a bit hard. however, i think censorship resistance would be valued more than mev as a whole solana ecosystem's interest. so, put differently, social consensus is need to move to scramble tx.

> I guess, one way to force validators to accept scrambled transactions is to leave them no other alternative 

yeah, this can be enforced to some extent even if we'll support normal (non-scrambled) transaction indefinitely. namely, vote txes must be made scrambled for inflation reward.

> What insentives validators have to keep their de-scrambling hardware (say, latest Intel CPUs) up to date ? I guess, they have to in order to be able to participate in consensus, if their CPU gets slow (compared to the latest targeted value, which should be updated every once in a while) it's same as downtime.

your reasoning is correct: opportunity cost for inflation reward or transaction fee necessitates prompt and periodic hw re-investment.